cross-dom-comm/README.md

```diff
@@ -0,0 +1,237 @@
+# How can a contract on one layer control a contract on the other
+
+[![Discord](https://img.shields.io/discord/667044843901681675.svg?color=768AD4&label=discord&logo=https%3A%2F%2Fdiscordapp.com%2Fassets%2F8c9701b98ad4372b58f13fd9f65f966e.svg)](https://discord.com/channels/667044843901681675)
+[![Twitter Follow](https://img.shields.io/twitter/follow/optimismPBC.svg?label=optimismPBC&style=social)](https://twitter.com/optimismPBC)
+
+This tutorial teaches you how to write a contract on one layer (either L1 or L2) that sends a message to call a function on a contract on the other layer. 
+[You can read more details about this process here](https://community.optimism.io/docs/developers/bridge/messaging/).
+
+## Seeing it in action
+
+To show how this works we installed [a slightly modified version of HardHat's `Greeter.sol`](contracts/Greeter.sol) on both L1 Kovan and Optimistic Kovan
+
+
+| Network | Greeter address  |
+| ------- | ---------------- |
+| Kovan (L1) | [0x11fB328D5Bd8E27917535b6d40b881d35BC39Be0](https://kovan.etherscan.io/address/0x11fB328D5Bd8E27917535b6d40b881d35BC39Be0) |
+| Optimistic Kovan (L2) | [0xD4c204223d6F1Dfad0b7a0b05BB0bCaB6665e0c9](https://kovan-optimistic.etherscan.io/address/0xD4c204223d6F1Dfad0b7a0b05BB0bCaB6665e0c9) |
```

Comment:
What happens if multiple people try this tutorial at the same time? Will stuff break?

Reply:
Not very much. If multiple people use it then they might see the wrong greeting (the one sent by the other person). I'll add a note to that effect.